### ISSUE TITLE
CA1062 not firing if '?.' is used on the previous line

### ISSUE THREAD

**Issue by mavasani** [dotnet/roslyn-analyzers#3531]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v3.0.0-beta3.final (latest pre-release)

#### Diagnostic ID

Example: [CA1062](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1062?view=vs-2019)

#### Repro steps

```csharp
namespace CA1062Test
{
    public static class Thing
	{
		public static void Fn(int[] items)
		{
			var l1 = items?.Length;

			// This should fire a CA1062, but it does not.
			var l2 = items.Length;
		}
	}
}
```

#### Expected behavior
CA1062 on line declaring `l2`

#### Actual behavior
No CA1062

**Comment by mavasani** [comment]
Reported at https://developercommunity.visualstudio.com/content/problem/945645/ca1062-only-appears-to-be-detected-for-the-first-o.html

**Comment by Youssef1313** [comment]
@Evangelink You may be interested in this. Similar to "ReDim" issue you worked on recently.

**Comment by Evangelink** [comment]
I have just made a test and can confirm this issue is still present. I did some debugging to try to understand what's going-on and I am having some problem to understand the flow of actions, from a theoretical point of view we should have something like this:

`var l1 = items?.Length;`
=> 1 path where `items` has `NotNull` constraint and 1 path where `items` has `Null` constraint.

`var l2 = items.Length;`
=> when reaching from `NotNull` path then learn nothing new, when reaching from `Null` constraint either stop exploration or continue with no new learning.

Doing the debugging I can see that we reach the following methods `VisitIsNullOperation`, `VisitFlowCaptureRef` and `VisitInvocation` but I don't seem to find where we reach with the `Null` path.
